src/praisonai-bot/praisonai_bot/bots/_session.py

```
@@ -34,6 +34,80 @@
+# When a durable store genuinely cannot be initialised the boundary that owns
+# it records a *durability-degraded* fact here instead of only logging a
+# warning, so the gateway can surface it through the surfaces operators already
+# read (``gateway doctor`` / ``gateway status``) — the same treatment channel
+# credential degradation already gets. This reuses the core degraded-owner
+# vocabulary (``owner_kind="gateway"``) so no new protocol is introduced; the
+# gateway's ``health()`` merges this process-local registry alongside its own.
+# --------------------------------------------------------------------------
+try:
+    from praisonaiagents.gateway import DegradedCapabilityRegistry as _DegRegistry
+    _DURABILITY_REGISTRY: Optional[Any] = _DegRegistry()
+except Exception:  # pragma: no cover - core registry unavailable
```

Ruff (0.16.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)